test/drivers/test_rsvd.cc

```diff
@@ -17,91 +17,118 @@ class TestRSVD : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// General test for RSVD:
-    /// Computes the decomposition factors, then checks A-U\Sigma\transpose{V}.
-    template <typename T, typename RNG>
-    static void test_RSVD1_general(int64_t m, int64_t n, int64_t k, int64_t p, int64_t block_sz, T tol, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state) {
-        
-        printf("|==================================TEST QB2 GENERAL BEGIN==================================|\n");
-
-        // For running QB
-        std::vector<T> A(m * n, 0.0);
-        RandLAPACK::util::gen_mat_type(m, n, A, k, state, mat_type);
-
-        int64_t size = m * n;
+    template <typename T>
+    struct RSVDTestData {
 
+        std::vector<T> A;
         // For results comparison
-        std::vector<T> A_hat(size, 0.0);
-        std::vector<T> A_hat_buf (m * k, 0.0);
-        std::vector<T> A_k(size, 0.0);
-        std::vector<T> A_cpy (m * n, 0.0);
+        std::vector<T> A_hat;
+        std::vector<T> A_hat_buf;
```

Comment:
Please use a different variable name here.

Reply:
Switched to a more descriptive one.